python/pulumi_aws_infra/network.py

```diff
@@ -114,7 +114,7 @@ def _create_from_arguments(self, name, number_of_availability_zones):
         :return: None
         """
         number_of_availability_zones = number_of_availability_zones or 2
-        if number_of_availability_zones < 1 or number_of_availability_zones > 4:
+        if number_of_availability_zones < 1 or number_of_availability_zones >= 4:
```

Comment:
This is driven by the AZs available in a region, right?  Or is there some other limitation here?

Reply:
Yes, that is true.

Reply:
Opened https://github.com/pulumi/pulumi-aws-infra/issues/21 to track potentially changing this to compute it from the target region.  It's not entirely trivial though, as we would then need to make an async call (which will likely in the future be a call that returns an `Output<T>`) before running the loop that creates each resource.  Interestingly, in Python that may not be as big a deal as it is in JavaScript - but I suspect we'll want to be consistent here across the two.